aiosql/adapters/generic.py

```diff
@@ -1,79 +1,84 @@
 from contextlib import contextmanager
 
-from ..patterns import var_pattern
 
+class GenericAdapter:
+    """
+    Generic AioSQL Adapter suitable for `named` parameter style and no with support.
+    """
 
-def replacer(match):
-    gd = match.groupdict()
-    if gd["dblquote"] is not None:
-        return gd["dblquote"]
-    elif gd["quote"] is not None:
-        return gd["quote"]
-    else:
-        return f'{gd["lead"]}%({gd["var_name"]})s{gd["trail"]}'
-
-
-class PsycoPG2Adapter:
     @staticmethod
     def process_sql(_query_name, _op_type, sql):
-        return var_pattern.sub(replacer, sql)
+        return sql
 
     @staticmethod
     def select(conn, _query_name, sql, parameters, record_class=None):
-        with conn.cursor() as cur:
+        cur = conn.cursor()
+        try:
             cur.execute(sql, parameters)
             results = cur.fetchall()
             if record_class is not None:
-                column_names = [c.name for c in cur.description]
+                column_names = [c[0] for c in cur.description]
                 results = [record_class(**dict(zip(column_names, row))) for row in results]
+        finally:
+            cur.close()
         return results
 
     @staticmethod
     def select_one(conn, _query_name, sql, parameters, record_class=None):
-        with conn.cursor() as cur:
+        cur = conn.cursor()
+        try:
             cur.execute(sql, parameters)
             result = cur.fetchone()
             if result is not None and record_class is not None:
-                column_names = [c.name for c in cur.description]
+                column_names = [c[0] for c in cur.description]
                 result = record_class(**dict(zip(column_names, result)))
+        finally:
+            cur.close()
         return result
 
     @staticmethod
     def select_value(conn, _query_name, sql, parameters):
-        with conn.cursor() as cur:
+        cur = conn.cursor()
+        try:
             cur.execute(sql, parameters)
             result = cur.fetchone()
+        finally:
+            cur.close()
         return result[0] if result else None
 
     @staticmethod
     @contextmanager
     def select_cursor(conn, _query_name, sql, parameters):
-        with conn.cursor() as cur:
-            cur.execute(sql, parameters)
+        cur = conn.cursor()
```

Comment:
What was the motivation behind using a try/finally block in these methods instead of entering the cursor context?

Reply:
This is because the *generic* adapter has to work with other drivers which do not necessarily have a working cursor context, and I wanted to have as much code in common as possible. AFAICR the either SQLite or MySQL does not support a context.